### PR TITLE
REST API: get site endpoint.

### PR DESCRIPTION
**Pull request by lezama**
Merge from wpcom

Add SAL, Site Abstraction Layer

cc @gravityrail

### CODE REVIEW

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 1]
```diff
@@ -22,13 +22,59 @@ class WPCOM_JSON_API_GET_Site_Endpoint extends WPCOM_JSON_API_Endpoint {
...
+			'image_medium_height',
+			'image_large_width',
+			'image_large_height',
+			'permalink_structure',
+			'post_formats',
+			'default_post_format',
+			'default_category',
+			'allowed_file_types',
+			'show_on_front',
+			/** This filter is documented in modules/likes.php */
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: A filter seems to be missing here.

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 2]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
-		 */
-		$response_format = apply_filters( 'sites_site_format', self::$site_format );
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$this->site = WPCOM_Platform::get_site( $blog_id );
+		} else {
+			$this->site = WPORG_Platform::get_site( $blog_id );
+		}
 
-		$is_user_logged_in = is_user_logged_in();
+		$default_fields = array_keys( apply_filters( 'sites_site_format', self::$site_format ) );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Could you add inline docs here?

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 3]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
 
-		$visible = array();
+		$response_keys = is_array( $this->fields_to_include ) ?
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Would it be possible to reformat this so it's more broken down and readable?

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 4]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
+		return $response;
+	}
 
-			// refactoring to change parameter to locale in 1.2
-			if ( $lang_or_locale = $this->process_locale( $key, $is_user_logged_in ) ) {
-				$response[$key] = $lang_or_locale;
-				continue;
-			}
+	protected function render_response_key( $key, &$response, $is_user_logged_in ) {
+		do_action( 'pre_render_site_response_key', $key );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Missing docblock

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 5]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
+		}
 
-				$is_redirect = false;
-
-				if ( function_exists( 'get_primary_domain_mapping_record' ) ) {
-					if ( get_primary_domain_mapping_record()->type == 1 ) {
-						$is_redirect = true;
-					}
-				}
+		do_action( 'post_render_site_response_key', $key );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Missing docblock

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 6]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
-					$wordads = has_any_blog_stickers( array( 'wordads-approved', 'wordads-approved-misfits' ), $blog_id );
-				}
+		$options = array();
 
-				$publicize_permanently_disabled = false;
-				if ( function_exists( 'is_publicize_permanently_disabled' ) ) {
-					$publicize_permanently_disabled = is_publicize_permanently_disabled( $blog_id );
-				}
+		// small optimisation - don't recalculate
+		$all_options = apply_filters( 'sites_site_options_format', self::$site_options_format );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Missing docblock

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 7]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
+				case 'default_category' :
+					$options[ $key ] = (int) get_option( 'default_category' );
+					break;
+				case 'allowed_file_types' :
+					$options[ $key ] = $this->site->allowed_file_types();
+					break;
+				case 'show_on_front' :
+					$options[ $key ] = get_option( 'show_on_front' );
+					break;
+				/** This filter is documented in modules/likes.php */
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: You'll need to move the comment right above the filter declaration so the parser can see it.

File: json-endpoints/class.wpcom-json-api-get-site-endpoint.php [hunk 8]
```diff
@@ -54,430 +119,273 @@ function callback( $path = '', $blog_id = 0 ) {
...
-			return (string) get_bloginfo( 'language' );
-		}
-		return false;
+		$this->site->after_render_options( $response['options'] );
 	}
 
-	function force_http( $url, $scheme, $orig_scheme ) {
-		return preg_replace('/^https:\/\//', 'http://', $url, 1 );
+	protected function build_meta_response( &$response ) {
+		$xmlrpc_scheme = apply_filters( 'wpcom_json_api_xmlrpc_scheme', parse_url( get_option( 'home' ), PHP_URL_SCHEME ) );
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-get-site-endpoint.php]: Missing docblock

**Comment by lezama** [comment]
Closed in favor of https://github.com/Automattic/jetpack/pull/3555

**Pull request closed without merging**